deeplearning/ml4pl/poj104/dataset.py

```diff
@@ -871,42 +901,32 @@ def process(self):
         data_list = []
         for i in tqdm.tqdm(range(num_files)):
             filename = input_files[i]
-            dat = dataset[i]
-            if filename[:3] == "npb":
-                # concatenate data set size
-                filename += '_' + str(dat)
-
-            # Updated from legacy
-            #file = Path(self.root) / 'kernels_ir_programl' / (filename + '.data.p')
-            file = Path(self.root) / 'kernels_ir' / (filename + '.ll.p')
-            
-            if file.exists():
-                # legacy
-                # load preprocessed data (without labels etc.)
-                #with open(file, 'rb') as f:
-                #    data = pickle.load(f)
-                # new
-                g = load(file)
-                data = nx2data(g)
-            else:
-                assert False, f'input file not found: {str(file)}. Did you run data preprocessing?'
 
-            # add data
-            data['y'] = torch.tensor([1]) if oracle[i] == 'GPU' else torch.tensor([0]) # CPU
-            data['aux_in'] = torch.tensor([[aux_transfer_size[i], aux_wg_size[i]]])
-            data['runtimes'] = torch.tensor([[runtime_cpu[i], runtime_gpu[i]]])
-            # TODO hacky fix for 
-            if hasattr(data, 'runtime'): delattr(data, 'runtime')
+            proto = load(filename, cdfg=self.cdfg)
+            data = nx2data(proto, vocabulary=vocab)
+
+            # graph2cdfg conversion drops the graph features, so we may have to 
+            # reload the graph.
+            if self.cdfg:
+                proto = load(filename)
+
+            # Add the features and label.
+            proto_features = proto.features.feature
+            data['y'] = torch.tensor(proto_features["devmap_label"].int64_list.value[0]).view(1)
+            data['aux_in'] = torch.tensor([
+                proto_features["transfer_bytes"].int64_list.value[0],
+                proto_features["wgsize"].int64_list.value[0],
+            ])
```

Comment:
Do we not get the runtimes for devmap?
I forgot if we even use them, or only accuracy. Did we say aggregated speedups are a meaningless metric?

Reply:
I didn't include runtimes in the exported features, so they're not here :( It would be easy to add them and compute a new dataset, so let me know if you want me to do that. My thinking was that we're only interested in accuracy